### PR TITLE
Guest agent: Add a case to check lifecycle's events

### PR DESCRIPTION
**Pull request by Yingshun**
This PR adds a case to check guest agent's lifecycle events.
BTW, it also adds guest_agent dirs to centralize related cases.

Signed-off-by: Yingshun Cui <yicui@redhat.com>
Case ID: VIRT-12689
**Depends on**: https://github.com/avocado-framework/avocado-vt/pull/3413
**Test results:** 
`(1/1) type_specific.io-github-autotest-libvirt.agent_lifecycle.events: PASS (68.68 s)`

### CODE REVIEW

**Comment by Yingshun** [comment]
@iccaszhulili Would you please help review this PR? Thanks!

File: libvirt/tests/src/guest_agent/agent_lifecycle.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        :param wait_for_event: wait until an event of the given type comes
+        :param event_type: type of the event
+        :param event_timeout: timeout for virsh event command
+        :param dargs: standardized function keywords
+        """
+        vm.prepare_guest_agent(**dargs)
+
+    @virsh.EventTracker.wait_event
+    def set_ga_status(name, wait_for_event=True,
+                      event_type='agent-lifecycle.*disconnected',
```
Comment (iccaszhulili) [inline libvirt/tests/src/guest_agent/agent_lifecycle.py:27]: The default attribute for set_state_guest_agent is to set guest agent as start, then why do we set the event_type as discconnected for default?

Comment (Yingshun) [reply libvirt/tests/src/guest_agent/agent_lifecycle.py]: Thanks, updated.

**Review by iccaszhulili — APPROVED** [review]

**Comment by iccaszhulili** [comment]
LGTM

**Review by nanli1 — APPROVED** [review]

**Review by chloerh — APPROVED** [review]

**Pull request merged**